### ISSUE TITLE
Add google_bigquery_analytics_hub_subscription resource

### ISSUE THREAD

**Issue by tanjt107** [hashicorp/terraform-provider-google#15144]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

I would like to request the addition of the `google_bigquery_analytics_hub_subscription` resource to the Terraform provider for Google Cloud Platform. This resource would allow users to create and manage subscriptions to data exchanges and listings in Analytics Hub.

Currently, the only way to create a subscription to a data exchange or listing in Terraform is to use console or API. However, this is a workaround, and it is not ideal. The `google_bigquery_analytics_hub_subscription` would have all of the features that are needed for creating and managing subscriptions, and it would be consistent with the other resources in the Terraform provider for Google Cloud Platform.

### New or Affected Resource(s)

* google_bigquery_analytics_hub_subscription

### Potential Terraform Configuration

```tf
resource "google_bigquery_analytics_hub_subscription" "default" {
  project-id = "my-project-id"
  location = "my-location"
  dataexchange0id = "my-dataexchange-id"
  listing-id = "my-listing-id"
}
```

### References

* https://cloud.google.com/bigquery/docs/analytics-hub-view-subscribe-listings

**Comment by DrFaust92** [comment]
Ill take a look as ive added support for hub and data exchange

**Comment by DrFaust92** [comment]
Actually having a deeper look. not sure this fits terraform, there is no real CRUD here. only:
https://cloud.google.com/bigquery/docs/reference/analytics-hub/rest/v1beta1/projects.locations.dataExchanges.listings/subscribe

There is no unsubscribe or get/list so there isnt a way to manage a state

**Comment by louisRDSC** [comment]
Hello @DrFaust92, it seems to have been added during august. is it what you were looking for?
https://cloud.google.com/bigquery/docs/reference/analytics-hub/rest/v1/projects.locations.subscriptions

**Comment by DrFaust92** [comment]
Yes, that might work. I'm going on vacation so I won't be looking at this for at least a few weeks.
I will add though that I hope it will work as it will need to call different endpoints and not the same func as other resources (I'm assuming it will just not might be trivial)

**Comment by terekete** [comment]
@DrFaust92 - given the new methods on the API can this be done ?

**Comment by Cidan** [comment]
Hi there,

I'm part of the BigQuery team directly, and I have a draft PR open for this [here](https://github.com/GoogleCloudPlatform/magic-modules/pull/9592) as well. I expect to be done in the next few days and send the PR in for full review after that.

Let me know if you'd like to merge efforts here, or if you want to continue on your own PR @sachinpro

Thanks!
